src/components/Technologies/Technologies.tsx

```diff
@@ -13,68 +18,151 @@ const searchOptions = {
-  const [selectedCategory, setSelectedCategory] = useState("all");
-  const [searchValue, setSearchValue] = useState("");
-
-  const categories = useMemo(() => {
-    const categoriesSet = data.reduce(
-      (accumulator, technology) => {
-        const categories = technology.fields.category || [];
-        categories.forEach((category) => {
-          accumulator.add(category);
-        });
-        return accumulator;
-      },
-      new Set(["all"])
-    );
-    return new Set([...categoriesSet].sort());
-  }, [data]);
+interface SearchFilter {
+  name: string;
+  values: any;
+}
 
-  const technologies = useMemo(() => {
-    let technologies = [...data].sort((pack1, pack2) => {
-      const category1 = pack1.fields.category[0];
-      const category2 = pack2.fields.category[0];
+export default function Technologies({ data, repositories }: TechnologiesProps) {
+  const { isDarkTheme } = useColorMode();
+  const { defaultAlgorithm, darkAlgorithm } = theme;
+  const [selectedFilters, setSelectedFilters] = useState<{ category: string[], registries: string[], cloudTypes: string, additionalFilters: string[] }>({ category: [], registries: [], cloudTypes: "", additionalFilters: [] })
+  const [searchValue, setSearchValue] = useState<string>("");
 
-      if (category1 < category2) {
-        return -1;
+  const filteredTechCards = useMemo(() => {
+    let _selectedFilters = [] as SearchFilter[];
+    Object.keys(selectedFilters).forEach((key) => {
+      const _values = selectedFilters[key as keyof typeof selectedFilters];
+      if (!_values || _values.length === 0) {
+        return;
       }
-
-      if (category1 > category2) {
-        return 1;
+      if (key === "additionalFilters") {
+        const additionalFilters = Array.isArray(_values) ? _values.reduce((acc, _value) => {
+          acc.push({
+            name: _value,
+            values: true
+          });
+          return acc;
+        }, [] as SearchFilter[]) : [];
+        _selectedFilters = _selectedFilters.concat(additionalFilters);
+      } else {
+        _selectedFilters.push({
+          name: key,
+          values: _values
+        });
       }
-
-      return 0;
     });
-
+    let filteredCards: any[] = [];
+    filteredCards = data.filter((techCard) => {
+      const card = techCard.fields;
+      return _selectedFilters.every((condition) => {
+        if (condition.name === "category") {
+          return condition.values.includes(card.packType);
+        } else if (condition.name === "registries") {
+          return condition.values.some((registry: string) => card.registries.includes(registry));
+        } else if (condition.name === "cloudTypes") {
+          return card.cloudTypes.includes("all") || card.cloudTypes.includes(condition.values);
+        } else if (["verified", "community"].includes(condition.name)) {
+          return card[condition.name as keyof FrontMatterData];
+        } else {
+          return true;
+        }
+      })
+    })
```

Comment:
doesn't look like it's done like I asked.
this function knows how to interpret conditions based on the name. so it knows more than it should.
This means that when I add a condition somewhere... then I have to add a "interpreter" here as well. This breaks coupling principles.